src/hooks/useTodos.ts

```diff
@@ -0,0 +1,127 @@
+import { useState, useEffect, useCallback } from 'react';
+import { addTodo, deleteTodo, getTodos, updateTodo } from '../api/todos';
+import { Todo } from '../types/Todo';
+import { ErrorMessages } from '../types/ErrorMessages';
+
+export const useTodos = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessages>(
+    ErrorMessages.NO_ERROR,
+  );
+  const [isReceivingAnswer, setIsReceivingAnswer] = useState(false);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [editedTodoId, setEditedTodoId] = useState<number | null>(null);
+  const [newTitle, setNewTitle] = useState('');
+
+  const handleErrorReset = useCallback(() => {
+    setErrorMessage(ErrorMessages.NO_ERROR);
+  }, [setErrorMessage]);
+
+  const handleErrorMessage = (error: ErrorMessages) => {
+    setErrorMessage(error);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => handleErrorMessage(ErrorMessages.UNABLE_TO_LOAD_TODO));
+  }, []);
```

Comment:
It's better to move the logic inside the `useEffect `hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep `useEffect `focused on side effects.